src/main/java/readAndWrite/MenuSingleRow.java

```diff
@@ -0,0 +1,38 @@
+package readAndWrite;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class MenuSingleRow {
+
+    private int menuRowNumber;
+    private String menuRowDescription;
+    private Map<String, Integer> availableOptionToAnswerOrType;
+
+    public MenuSingleRow(int menuRowNumber, String menuRowDescription, Map<String, Integer> availableOptionToAnswerOrType) {
+        this.menuRowNumber = menuRowNumber;
+        this.menuRowDescription = menuRowDescription;
+        if(availableOptionToAnswerOrType != null) {
+            this.availableOptionToAnswerOrType = new LinkedHashMap<>(availableOptionToAnswerOrType);
+        }else {
+            this.availableOptionToAnswerOrType = new LinkedHashMap<>();
+        }
+        this.availableOptionToAnswerOrType.put("q", 0);
+    }
+
+    public int getMenuRowNumber() {
+        return menuRowNumber;
+    }
+
+    public String getMenuRowDescription() {
+        return menuRowDescription;
+    }
+
+    public Map<String, Integer> getAvailableOptionToAnswerOrType() {
+        return new LinkedHashMap<>(availableOptionToAnswerOrType);
```

Comment:
You have the map already created and every time somebody calls this method a new map will be created. Why?